File: VB/WpfApplication1/RestrictionsViewModel.vb
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm.POCO

Namespace WpfApplication1

    Public Class RestrictionsViewModel

        Private ReadOnly data As SampleDataGenerator = New SampleDataGenerator()

        Protected Sub New()
            AppointmentsPerDay = 10
            DayCount = 2
            ResourceCount = 3
        End Sub

        Public Overridable ReadOnly Property Appointments As ObservableCollection(Of AppointmentData)
            Get
                Return data.Appointments
            End Get
        End Property

        Public Overridable ReadOnly Property Resources As ObservableCollection(Of ResourceData)
            Get
                Return data.Resources
            End Get
        End Property

        Public Overridable Property AppointmentsPerDay As Integer

        Public Overridable Property DayCount As Integer

        Public Overridable Property ResourceCount As Integer

        Protected Sub OnDayCountChanged()
            data.SetUp(DayCount, ResourceCount, AppointmentsPerDay)
        End Sub

        Protected Sub OnResourceCountChanged()
            data.SetUp(DayCount, ResourceCount, AppointmentsPerDay)
        End Sub

        Public Shared Function Create() As RestrictionsViewModel
            Return ViewModelSource.Create(Function() New RestrictionsViewModel())
        End Function
    End Class
End Namespace

File: CS/WpfApplication1/RestrictionsViewModel.cs
using System;
using System.Collections.ObjectModel;
using System.Linq;
using DevExpress.Mvvm.POCO;
using DevExpress.XtraScheduler;

namespace WpfApplication1 {
    public class RestrictionsViewModel {
        readonly SampleDataGenerator data = new SampleDataGenerator();
        protected RestrictionsViewModel() {
            AppointmentsPerDay = 10;
            DayCount = 2;
            ResourceCount = 3;
        }

        public virtual ObservableCollection<AppointmentData> Appointments { get { return data.Appointments; } }
        public virtual ObservableCollection<ResourceData> Resources { get { return data.Resources; } }
        public virtual int AppointmentsPerDay { get; set; }
        public virtual int DayCount { get; set; }
        public virtual int ResourceCount { get; set; }
        protected void OnDayCountChanged() {
            data.SetUp(DayCount, ResourceCount, AppointmentsPerDay);
        }
        protected void OnResourceCountChanged() {
            data.SetUp(DayCount, ResourceCount, AppointmentsPerDay);
        }
        public static RestrictionsViewModel Create() {
            return ViewModelSource.Create(() => new RestrictionsViewModel());
        }
    }
}
